### PR TITLE
Bug 1513708 - BMO OAuth2 DB tables should be using proper primary key ID columns

### PR DESCRIPTION
**Pull request by dklawren**
t/mojo-oauth2.t passes at this point. Still want to finish the migration code.

### CODE REVIEW

File: Bugzilla/DB/Schema.pm [hunk 1]
```diff
@@ -1885,16 +1889,17 @@ use constant ABSTRACT_SCHEMA => {
...
-        {FIELDS => ['auth_code', 'scope_id'], TYPE => 'UNIQUE'},
+        {FIELDS => ['auth_code_id', 'scope_id'], TYPE => 'UNIQUE'},
     ]
   },
 
   oauth2_access_token => {
     FIELDS => [
-      access_token  => {TYPE => 'varchar(255)', NOTNULL => 1, PRIMARYKEY => 1},
+      id            => {TYPE => 'MEDIUMSERIAL', NOTNULL => 1, PRIMARYKEY => 1},
+      access_token  => {TYPE => 'varchar(255)', NOTNULL => 1},
```
Comment (dylanwh) [inline Bugzilla/DB/Schema.pm]: access_token_jti => { TYPE => 'varchar(255)', NOTNULL => 1 }

File: Bugzilla/DB/Schema.pm [hunk 2]
```diff
@@ -1885,16 +1889,17 @@ use constant ABSTRACT_SCHEMA => {
...
       refresh_token => {TYPE => 'varchar(255)'},
```
Comment (dylanwh) [inline Bugzilla/DB/Schema.pm]: refresh_token_jti => { TYPE => 'varchar(255)', NOTNULL => 1 }

**Comment by dklawren** [comment]
Ready for proper review

**Comment by dklawren** [comment]
Scratch this whole mess :( The JWT conversion will remove the need for this PR since quite a few tables are going to be removed.

**Pull request closed without merging**